torch/_dynamo/side_effects.py

```diff
@@ -186,8 +186,11 @@ def check_allowed_side_effect(self, item):
                 "unintended variable modifications."
             )
         if not is_side_effect_safe(item.mutation_type):
-            unimplemented(
-                "HigherOrderOperator: Mutating a variable not in the current scope (SideEffects)"
+            unimplemented_v2(
+                gb_type="HigherOrderOperator: Mutating a variable not in the current scope (SideEffects)",
+                context="",
+                explanation="This is not supported.",
+                hints=[],
```

Comment:
Can we do something like: 
"Mutating a variable not in the current scope inside of {HOP_NAME} is not supported"

Reply:
Will have to figure out how to plumb the HOP name in

Reply:
Doesn't need to happen in this PR (can you file an issue for a follow-up?)